main_test.go

```diff
@@ -79,3 +79,23 @@ func ExampleVersion() {
 	// Output:
 	// chainlink.test version unset@unset
 }
+
+func ExampleWithdraw() {
+	tc, cleanup := cltest.NewConfig()
+	defer cleanup()
+	testClient := &cmd.Client{
+		Renderer:               cmd.RendererTable{Writer: ioutil.Discard},
+		Config:                 tc.Config,
+		AppFactory:             cmd.ChainlinkAppFactory{},
+		KeyStoreAuthenticator:  cmd.TerminalKeyStoreAuthenticator{Prompter: &cltest.MockCountingPrompter{}},
+		FallbackAPIInitializer: &cltest.MockAPIInitializer{},
+		Runner:                 cmd.ChainlinkRunner{},
+		HTTP:                   cltest.NewMockAuthenticatedHTTPClient(tc.Config),
+	}
+
+	Run(testClient,
+		"chainlink.test", "withdraw",
+		"0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef", "1234",
+		"--from-oracle-contract-address=" +
+			"0x3141592653589793238462643383279502884197")
```

Comment:
Two quick points:

1. Let's throw a case sensitive address in the example to convey that we care about EIP55.
2. When using examples, a comment right below `Run` is used as golden output to assert that the output of the command run matches the quoted text. You're not doing that here, which makes the example different with the other ones. I don't really think that's bad in and of itself, but this `Run(...)` like will actually error out. Without an assertion to ensure it works, this will get stale quickly.

If you wanted to go the extra mile, you could do the appropriate setup with `ethmock` to have it succeed and match against correct output. As it is now, I would ✂️ cut it.

Reply:
Thanks. I got this far with it:

```go
func ExampleWithdraw() {
        tc, cleanup := cltest.NewConfigWithPrivateKey()
        defer cleanup()
        app, cleanup := cltest.NewApplicationWithConfigAndKeyStore(tc)
        defer cleanup()

        testClient := &cmd.Client{
                Renderer:               cmd.RendererTable{Writer: ioutil.Discard},
                Config:                 tc.Config,
                AppFactory:             cmd.ChainlinkAppFactory{},
                KeyStoreAuthenticator:  cmd.TerminalKeyStoreAuthenticator{Prompter: &cltest.MockCountingPrompter{}},
                FallbackAPIInitializer: &cltest.MockAPIInitializer{},
                Runner:                 cmd.ChainlinkRunner{},
                HTTP:                   cltest.NewMockAuthenticatedHTTPClient(tc.Config),
        }

        hash := cltest.NewHash()
        ethMock := app.MockEthClient()
        sentAt := "0x5BA0"
        nonce := "0x100"

        ethMock.Context("app.Start()", func(ethMock *cltest.EthMock) {
                ethMock.Register("eth_getTransactionCount", nonce)
        })

        ethMock.Context("manager.CreateTx#1", func(ethMock *cltest.EthMock) {
                ethMock.Register(
                        "eth_call", "0xDE0B6B3A7640000")
                ethMock.Register("eth_sendRawTransaction", hash)
                ethMock.Register("eth_blockNumber", sentAt)
        })

        _ = app.Start()

        Run(testClient,
                "chainlink.test", "withdraw",
                "0xDeaDbeefdEAdbeefdEadbEEFdeadbeEFdEaDbeeF", "1234",
                "--from-oracle-contract-address=" +
                        "0x3141592653589793238462643383279502884197")
        // Output:                                                                                                                                                                                                                                                              
        // foo                                                                                                                                                                                                                                                                  
}
```

...but it's failing with `401 Unauthorized` when it tries to hit the `/v2/withdrawals` endpoint. I guess if it's not clear how to fix that it's best to just cut it for now.

Reply:
I switched your setup code to use our `NewClientAndRenderer` helper:

```golang
	app, cleanup := cltest.NewApplicationWithKeyStore()
	defer cleanup()

	testClient, _ := app.NewClientAndRenderer()
```

Check out the latest commit.

Reply:
Thanks, Dmitri. I decided a full integration test wasn't worth the trouble. I added a test with the same CL arguments in `remote_client_test.go/TestClient_WithdrawFromSpecifiedContractAddress`. I think it's complete, now. PTAL, when you get a chance.